edk2toolext/capsule/capsule_helper.py

```diff
@@ -0,0 +1,162 @@
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+##
+
+
+import uuid
+import os
+# TODO: Replace this step with a new function from within
+#       FmpAuthHeaderClass().
+import struct
+
+from edk2toollib.windows.capsule import inf_generator, cat_generator
+from edk2toollib.uefi.uefi_capsule_header import UefiCapsuleHeaderClass
+from edk2toollib.uefi.fmp_capsule_header import FmpCapsuleHeaderClass, FmpCapsuleImageHeaderClass
+from edk2toollib.uefi.fmp_auth_header import FmpAuthHeaderClass
+from edk2toollib.uefi.edk2.fmp_payload_header import FmpPayloadHeaderClass
+
+
+def get_capsule_file_name(capsule_options):
+    return f"{capsule_options['fw_name']}_{capsule_options['fw_version_string']}.bin"
+
+
+def get_normalized_version_string(version_string):
+    # 19H1 HLK requires a 4 digit version string, or it will fail
+    while (version_string.count('.') < 3):
+        version_string += '.0'
+    return version_string
+
+
+def get_default_arch():
+    return 'amd64'
+
+
+def get_default_os_string():
+    return 'Win10'
+
+
+def build_capsule(capsule_data, capsule_options, signer_module, signer_options):
+    # Start building the capsule as we go.
+    # Create the FMP Payload and set all the necessary options.
+    fmp_payload_header = FmpPayloadHeaderClass()
+    fmp_payload_header.FwVersion = int(capsule_options['fw_version'], 16)
+    fmp_payload_header.LowestSupportedVersion = int(capsule_options['lsv_version'], 16)
+    fmp_payload_header.Payload = capsule_data
+
+    # Create the auth header and get ready to sign the data.
+    fmp_auth_header = FmpAuthHeaderClass()
+    # TODO: Allow caller to pass the monotonic count in capsule_options.
+    fmp_auth_header.MonotonicCount = 1
+    fmp_auth_header.FmpPayloadHeader = fmp_payload_header
+
+    # TODO: Replace this step with a new function from within
+    #       FmpAuthHeaderClass().
+    data_to_sign = fmp_payload_header.Encode()
+    data_to_sign = data_to_sign + struct.pack("<Q", fmp_auth_header.MonotonicCount)
+
+    # Sign the data and assign it to the cert data.
+    # TODO: Determine whether EKU is a signer option or a signature option.
+    # TODO: Figure out whether this sign_alg is correct for capsules.
+    signature_options = {
+        'sign_alg': 'pkcs12',
+        'hash_alg': 'sha256'
+    }
+    # Set or override OID.
+    signer_options['oid'] = '1.2.840.113549.1.7.2'
```

Comment:
would like to see this OID defined rather that number text used as it requires look up.